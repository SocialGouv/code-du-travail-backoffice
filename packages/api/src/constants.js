// TODO Fix this hack.
const COMMON_HEADERS = {
  "Access-Control-Allow-Headers": "Authorization, Content-Type",
  "Access-Control-Allow-Methods": "GET, OPTIONS",
  "Access-Control-Allow-Origin": "*",
  "Content-Type": "application/json",
};

module.exports = {
  COMMON_HEADERS,
};
