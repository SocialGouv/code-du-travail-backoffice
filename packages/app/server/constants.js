const USER_ROLE = {
  ADMINISTRATOR: "administrator",
  CONTRIBUTOR: "contributor",
  REGIONAL_ADMINISTRATOR: "regional_administrator",
};

module.exports = { USER_ROLE };
