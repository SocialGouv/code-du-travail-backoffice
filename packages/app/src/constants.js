import capitalize from "./helpers/capitalize";

export const ANSWER_GENERIC_REFERENCE = {
  LABOR_CODE: "labor_code",
  NATIONAL_AGREEMENT: "national_agreement",
};
export const ANSWER_STATE = {
  DRAFT: "draft",
  PENDING_REVIEW: "pending_review",
  TO_DO: "todo",
  UNDER_REVIEW: "under_review",
  VALIDATED: "validated",
};
export const ANSWER_STATE_LABEL = {
  draft: "en cours de rédaction",
  pending_review: "en attente de validation",
  todo: "à rédiger",
  under_review: "en cours de validation",
  validated: "validée",
};
export const ANSWER_STATES = Object.keys(ANSWER_STATE_LABEL);
export const ANSWER_STATE_OPTIONS = ANSWER_STATES.map(state => ({
  label: capitalize(ANSWER_STATE_LABEL[state]),
  value: state,
}));

export const ANSWER_REFERENCE_CATEGORY = {
  AGREEMENT: "agreement",
  LABOR_CODE: "labor_code",
};

export const AREA_CATEGORY_LABEL = {
  department: "Département",
  overseas_collectivity: "Collectivité d'Outre-Mer",
  region: "Région",
  sui_generis_collectivity: "Collectivité Sui Generis",
};

export const LOG_ACTION_LABEL = {
  delete: "SUPPRESSION",
  patch: "MODIFICATION",
  post: "INSERTION",
};

export const SESSION = {
  ANSWERS_FILTERS: "history.answers.filters",
  GENERIC_ANSWERS_FILTERS: "history.genericAnswers.filters",
};

export const USER_ROLE = {
  ADMINISTRATOR: "administrator",
  CONTRIBUTOR: "contributor",
  REGIONAL_ADMINISTRATOR: "regional_administrator",
};
