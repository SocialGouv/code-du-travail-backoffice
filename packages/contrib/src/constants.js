export const ANSWER_STATE = {
  TODO: "todo",
  DRAFT: "draft",
  PENDING_REVIEW: "pending_review",
  VALIDATED: "validated"
};
export const ANSWER_STATE_LABEL = {
  todo: "À rédiger",
  draft: "Brouillon",
  pending_review: "En attente de validation",
  validated: "Validée"
};

export const ZONE_CATEGORY_LABEL = {
  department: "Département",
  overseas_collectivity: "Collectivité d'Outre-Mer",
  region: "Région",
  sui_generis_collectivity: "Collectivité Sui Generis"
};
