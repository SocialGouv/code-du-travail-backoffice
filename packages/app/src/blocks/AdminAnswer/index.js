import styled from "@emotion/styled";
import React from "react";
import { Flex } from "rebass";

import Checkbox from "../../elements/Checkbox";
import Idcc from "../../elements/Idcc";
import excerpt from "../../helpers/excerpt";
import customMoment from "../../libs/customMoment";

const Container = styled(Flex)`
  margin-top: 1rem;
  user-select: none;
`;

const Top = styled(Flex)`
  font-size: 0.75rem;
  font-weight: 700;
  margin-bottom: 0.25rem;
  padding-left: 3.25rem;
  text-transform: uppercase;
`;
const TopState = styled.span`
  color: var(--color-shadow);
`;
const TopAuthor = styled.span`
  color: var(--color-blue-sapphire);
`;
const TopPublished = styled.span`
  background-color: var(--color-blue-sapphire);
  color: white;
  font-size: 0.625rem;
  opacity: ${({ disabled }) => (disabled ? 0.1 : 1)};
  padding: 0.125rem 0.25rem;
`;

const Content = styled(Flex)`
  background-color: white;
  border: solid 1px var(--color-border);
  border-radius: 0.4rem;
  cursor: pointer;
  flex-grow: 1;
  padding: 0.75rem 0.5rem 0.75rem 0.75rem;
`;
const ContentQuestion = styled.div`
  color: black;
  font-weight: 600;
`;
const ContentExtract = styled.div`
  color: var(--color-text-gray);
  font-size: 0.875rem;
  margin-top: 0.5rem;
`;
const ContentExtractRed = styled(ContentExtract)`
  color: var(--color-text-red);
  font-weight: 700;
`;

import { ANSWER_STATE, ANSWER_STATE_LABEL } from "../../constants";

const AdminAnswerBlock = ({ data, isChecked, onCheck, onClick }) => {
  const {
    agreement,
    generic_reference,
    id,
    is_published,
    question,
    state,
    updated_at,
    user,
  } = data;

  const isGeneric = agreement === null;
  const isTodo = state === ANSWER_STATE.TO_DO;
  const value = [ANSWER_STATE.DRAFT, ANSWER_STATE.PENDING_REVIEW].includes(state)
    ? data.prevalue
    : data.value;

  return (
    <Container flexDirection="column">
      <Top justifyContent="space-between">
        <TopState>{ANSWER_STATE_LABEL[state]}</TopState>
        {![ANSWER_STATE.TO_DO, ANSWER_STATE.VALIDATED].includes(state) &&
          user !== undefined &&
          user !== null && (
            <TopAuthor>
              {state === ANSWER_STATE.DRAFT && `Rédigée par : ${user.name}`}
              {[ANSWER_STATE.PENDING_REVIEW, ANSWER_STATE.UNDER_REVIEW].includes(state) &&
                `Proposée par : ${user.name}`}
              {state !== ANSWER_STATE.UNDER_REVIEW &&
                `, ${customMoment(updated_at).tz("Europe/Paris").fromNow()}`}
            </TopAuthor>
          )}
        {state === ANSWER_STATE.VALIDATED && (
          <TopPublished disabled={!is_published}>Publiée</TopPublished>
        )}
      </Top>
      <Flex>
        {isChecked !== undefined && (
          <Checkbox
            isChecked={isChecked}
            onClick={() => onCheck(id)}
            withLeftMargin
            withRightMargin
          />
        )}
        <Content flexDirection="column" onClick={() => onClick(id)}>
          <Flex alignItems="baseline">
            {!isGeneric ? <Idcc code={agreement.idcc} name={agreement.name} /> : <Idcc />}
            <ContentQuestion>{`${question.index}) ${question.value}`}</ContentQuestion>
          </Flex>
          {!isTodo && generic_reference === null && (
            <ContentExtract>{excerpt(value)}</ContentExtract>
          )}
          {!isTodo && generic_reference === "labor_code" && (
            <ContentExtractRed>Renvoyé au Code du travail.</ContentExtractRed>
          )}
          {!isTodo && generic_reference === "national_agreement" && (
            <ContentExtractRed>Renvoyé à la convention collective nationale.</ContentExtractRed>
          )}
        </Content>
      </Flex>
    </Container>
  );
};

export default AdminAnswerBlock;
