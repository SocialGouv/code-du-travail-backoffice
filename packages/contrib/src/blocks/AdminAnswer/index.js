import React from "react";
import { Flex } from "rebass";
import styled from "styled-components";

import Idcc from "../../elements/Idcc";
import Checkbox from "../../elements/Checkbox";
import excerpt from "../../helpers/excerpt";
import customMoment from "../../libs/customMoment";

const Container = styled(Flex)`
  margin-top: 1rem;
  user-select: none;
`;

const Top = styled(Flex)`
  font-size: 0.75rem;
  font-weight: 700;
  margin-bottom: 0.25rem;
  text-transform: uppercase;
`;
const TopState = styled.span`
  color: var(--color-shadow);
`;
const TopAuthor = styled.span`
  color: var(--color-blue-sapphire);
`;

const Content = styled(Flex)`
  background-color: white;
  border: solid 1px var(--color-border);
  border-radius: 0.4rem;
  cursor: pointer;
  flex-grow: 1;
  padding: 0.75rem 0.5rem 0.75rem 0.75rem;
`;
const ContentQuestion = styled.div`
  color: black;
  font-weight: 600;
`;
const ContentExtract = styled.div`
  color: var(--color-text-gray);
  font-size: 0.875rem;
  margin-top: 0.5rem;
`;
const ContentExtractRed = styled(ContentExtract)`
  color: var(--color-text-red);
  font-weight: 700;
`;

import { ANSWER_STATE, ANSWER_STATE_LABEL } from "../../constants";

export default ({ data, isChecked, onCheck, onClick }) => {
  const isTodo = data.state === ANSWER_STATE.TO_DO;
  const value = [ANSWER_STATE.DRAFT, ANSWER_STATE.PENDING_REVIEW].includes(
    data.state
  )
    ? data.prevalue
    : data.value;

  return (
    <Container>
      {isChecked !== undefined && (
        <Checkbox
          icon={isChecked ? "check-square" : "square"}
          onClick={() => onCheck(data.id)}
        />
      )}
      <Flex flexDirection="column" width={1}>
        <Top justifyContent="space-between">
          <TopState>{ANSWER_STATE_LABEL[data.state]}</TopState>
          {data.user !== null && (
            <TopAuthor>
              {`Proposé par : ${data.user.name}, ${customMoment(data.updated_at)
                .tz("Europe/Paris")
                .fromNow()}`}
            </TopAuthor>
          )}
        </Top>
        <Content flexDirection="column" onClick={() => onClick(data.id)}>
          <Flex alignItems="baseline">
            <Idcc code={data.agreement_idcc} name={data.agreement_name} />
            <ContentQuestion>
              {`${data.question_index}) ${data.question_value}`}
            </ContentQuestion>
          </Flex>
          {!isTodo && data.generic_reference === null && (
            <ContentExtract>{excerpt(value)}</ContentExtract>
          )}
          {!isTodo && data.generic_reference === "labor_code" && (
            <ContentExtractRed>Renvoyé au Code du travail.</ContentExtractRed>
          )}
          {!isTodo && data.generic_reference === "national_agreement" && (
            <ContentExtractRed>
              Renvoyé à la convention collective nationale.
            </ContentExtractRed>
          )}
        </Content>
      </Flex>
    </Container>
  );
};
