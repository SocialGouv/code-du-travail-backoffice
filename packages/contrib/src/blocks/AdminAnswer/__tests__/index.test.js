import moment from "moment-timezone";
import React from "react";
import { fireEvent, render } from "@testing-library/react";

import AdminAnswer from "..";

import { ANSWER_STATE } from "../../../constants";

// Ignore styled-wrapped ReactTooltip className prop warning
console.warn = jest.fn();

const PROPS = {
  data: {
    id: "12345678-9abc-4def-0123-456789abcdef",
    generic_reference: null,
    state: ANSWER_STATE.DRAFT,
    updated_at: moment()
      .subtract(2, "hours")
      .tz("Europe/Paris"),
    prevalue: `Il m'est arrivé de sentir que tout allait bien pour moi, que<br>
            tout le monde était gai, et aussitôt l'idée me traversait<br>
            l'esprit qu'il ne se passerait plus rien et que tout était<br>
            absurde.`,
    agreement: {
      idcc: "1234",
      name: "An Agreement Name"
    },
    question: {
      index: 12,
      value: "A Question Value"
    },
    user: {
      name: "A User Name"
    }
  },
  onCheck: jest.fn()
};

describe("[Contrib] blocks/<AdminAnswer /> (Draft)", () => {
  const props = { ...PROPS };
  const λ = render(<AdminAnswer {...props} />);

  it("should match snapshot", () => {
    expect(λ.container).toMatchSnapshot();
  });

  it("should have called onCheck() with the expected param", () => {
    fireEvent.click(λ.getAllByRole("checkbox")[0]);

    expect(props.onCheck).toHaveBeenCalledWith(props.data.id);
  });
});

describe("[Contrib] blocks/<AdminAnswer /> (Draft falling back to LC)", () => {
  const props = {
    ...PROPS,
    data: {
      ...PROPS.data,
      generic_reference: "labor_code",
      value: ""
    }
  };
  const λ = render(<AdminAnswer {...props} />);

  it("should match snapshot", () => {
    expect(λ.container).toMatchSnapshot();
  });
});

describe("[Contrib] blocks/<AdminAnswer /> (Draft falling back to NA)", () => {
  const props = {
    ...PROPS,
    data: {
      ...PROPS.data,
      generic_reference: "national_agreement",
      value: ""
    }
  };
  const λ = render(<AdminAnswer {...props} />);

  it("should match snapshot", () => {
    expect(λ.container).toMatchSnapshot();
  });
});

describe("[Contrib] blocks/<AdminAnswer /> (Checked draft)", () => {
  const props = {
    ...PROPS,
    isChecked: true
  };
  const λ = render(<AdminAnswer {...props} />);

  it("should match snapshot", () => {
    expect(λ.container).toMatchSnapshot();
  });
});

describe("[Contrib] blocks/<AdminAnswer /> (Unchecked draft)", () => {
  const props = {
    ...PROPS,
    isChecked: false
  };
  const λ = render(<AdminAnswer {...props} />);

  it("should match snapshot", () => {
    expect(λ.container).toMatchSnapshot();
  });

  it("should have called onCheck() with the expected param", () => {
    fireEvent.click(λ.getAllByRole("checkbox")[0]);

    expect(props.onCheck).toHaveBeenCalledWith(props.data.id);
  });
});
