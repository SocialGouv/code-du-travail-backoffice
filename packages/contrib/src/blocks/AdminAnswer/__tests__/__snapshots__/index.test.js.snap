// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] blocks/<AdminAnswer /> (Checked draft) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <button
      class="sc-bdVaJa sc-EHOje Checkbox-sc-1jox9lb-0 efNukH Button-sc-1bpx84u-0 wNuyp"
      color="white"
      font-size="inherit"
      font-weight="bold"
      role="checkbox"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-check-square fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
        color="var(--color-lapis-lazuli)"
        data-icon="check-square"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          en cours de rédaction
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 dyxMzJ"
        >
          Il m'est arrivé de sentir que tout allait bien pour moi, que&lt;br&gt;
            tout le monde était…
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (Draft falling back to LC) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          en cours de rédaction
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 AdminAnswer__ContentExtractRed-sc-1257a91-7 hHWLgm"
        >
          Renvoyé au Code du travail.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (Draft falling back to NA) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          en cours de rédaction
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 AdminAnswer__ContentExtractRed-sc-1257a91-7 hHWLgm"
        >
          Renvoyé à la convention collective nationale.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (Draft) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          en cours de rédaction
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 dyxMzJ"
        >
          Il m'est arrivé de sentir que tout allait bien pour moi, que&lt;br&gt;
            tout le monde était…
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (Pending Review) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          en attente de validation
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 dyxMzJ"
        >
          Il m'est arrivé de sentir que tout allait bien pour moi, que&lt;br&gt;
            tout le monde était…
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (To Do) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          à rédiger
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (Unchecked draft) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <button
      class="sc-bdVaJa sc-EHOje Checkbox-sc-1jox9lb-0 efNukH Button-sc-1bpx84u-0 wNuyp"
      color="white"
      font-size="inherit"
      font-weight="bold"
      role="checkbox"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-square fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
        color="var(--color-lapis-lazuli)"
        data-icon="square"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          en cours de rédaction
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 dyxMzJ"
        >
          Il m'est arrivé de sentir que tout allait bien pour moi, que&lt;br&gt;
            tout le monde était…
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (Under Review) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          en cours de validation
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 dyxMzJ"
        >
          Il m'est arrivé de sentir que tout allait bien pour moi, que&lt;br&gt;
            tout le monde était…
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] blocks/<AdminAnswer /> (Validated) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminAnswer__Container-sc-1257a91-0 exoODn"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH eTLtBf"
      width="1"
    >
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Top-sc-1257a91-1 dgddaa"
      >
        <span
          class="AdminAnswer__TopState-sc-1257a91-2 RjFBy"
        >
          validée
        </span>
        <span
          class="AdminAnswer__TopAuthor-sc-1257a91-3 ecauMH"
        >
          Proposé par : A User Name, il y a 2 heures
        </span>
      </div>
      <div
        class="sc-bdVaJa sc-bwzfXH AdminAnswer__Content-sc-1257a91-4 httAcf"
      >
        <div
          class="sc-bdVaJa sc-bwzfXH dntGME"
        >
          <div
            class="Idcc__Container-owzimc-0 fWXTkt"
            currentitem="false"
            data-tip="An Agreement Name"
          >
            CCN: 
            1234
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            />
          </div>
          <div
            class="AdminAnswer__ContentQuestion-sc-1257a91-5 bTdchR"
          >
            12) A Question Value
          </div>
        </div>
        <div
          class="AdminAnswer__ContentExtract-sc-1257a91-6 dyxMzJ"
        >
          Il m'est arrivé de sentir que tout allait bien pour moi, que&lt;br&gt;
            tout le monde était…
        </div>
      </div>
    </div>
  </div>
</div>
`;
